### PR TITLE
[ci-visibility] Support parallel mode in cucumber

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?

* Add the necessary hooks to be able to run cucumber in parallel. 
* Rename `jest-worker` exporter to `test-worker`.
* Use `test-worker` exporter for cucumber as well. 

⚠️ ITR and EFD will not work right away with parallel mode. This will be tackled at another PR. 

ℹ️ Multiple tests failing because of node22 release. Being fixed in https://github.com/DataDog/dd-trace-js/pull/4267

### Motivation
Allow users to use parallel mode with cucumber: https://github.com/cucumber/cucumber-js/blob/main/docs/parallel.md. It used to be incompatible with dd-trace.

### Plugin Checklist

- [x] Unit tests.

### CODE REVIEW

File: integration-tests/cucumber/cucumber.spec.js [hunk 1]
```diff
@@ -123,112 +97,134 @@ versions.forEach(version => {
             isAgentless = reportMethod === 'agentless'
             envVars = isAgentless ? getCiVisAgentlessConfig(receiver.port) : getCiVisEvpProxyConfig(receiver.port)
           })
-          it('can run and report tests', (done) => {
```
Comment (juan-fernandez) [inline integration-tests/cucumber/cucumber.spec.js:126]: The changes in this file are just for running this test both in serial and parallel mode

File: packages/datadog-instrumentations/src/cucumber.js [hunk 2]
```diff
@@ -106,6 +115,43 @@ function getTestStatusFromRetries (testStatuses) {
...
+  return runtime.pickleIds.reduce((acc, pickleId) => {
+    const test = runtime.eventDataCollector.getPickle(pickleId)
+    if (acc[test.uri]) {
+      acc[test.uri].push(test)
+    } else {
+      acc[test.uri] = [test]
+    }
+    return acc
+  }, {})
+}
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/cucumber.js:153]: ℹ️ `getFilteredPickles` and `getPickleByFile` were hoisted but no other changes

File: packages/datadog-instrumentations/src/cucumber.js [hunk 3]
```diff
@@ -207,137 +248,67 @@ function pickleHook (PickleRunner) {
...
-  name: '@cucumber/cucumber',
-  versions: ['7.0.0 - 7.2.1'],
-  file: 'lib/runtime/pickle_runner.js'
-}, pickleHook)
-
-addHook({
-  name: '@cucumber/cucumber',
-  versions: ['>=7.3.0'],
-  file: 'lib/runtime/test_case_runner.js'
-}, testCaseHook)
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/cucumber.js:233]: ℹ️ these two hooks were just moved

File: packages/datadog-instrumentations/src/cucumber.js [hunk 4]
```diff
@@ -432,22 +404,106 @@ function getWrappedRunTest (runTestFunction) {
...
-      testSuiteFinishCh.publish(testSuiteStatus)
+      testSuiteFinishCh.publish({ status: testSuiteStatus, testSuitePath })
     }
 
     return runTestCaseResult
   }
 }
 
-// From 7.3.0 onwards, runPickle becomes runTestCase
+function getWrappedParseWorkerMessage (parseWorkerMessageFunction) {
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/cucumber.js:423]: ℹ️ `getWrappedParseWorkerMessage` is for intercepting messages from the workers and: 
* flush test data coming from the workers (workers now use the interprocess exporter)
* start / finish test suites

File: packages/datadog-instrumentations/src/cucumber.js [hunk 5]
```diff
@@ -469,3 +528,21 @@ addHook({
 
   return runtimePackage
 })
+
+// Only executed in parallel mode.
+// `getWrappedStart` generates session start and finish events
+// `getWrappedGiveWork` generates suite start events and sets pickleResultByFile (used by suite finish events)
+// `getWrappedParseWorkerMessage` generates suite finish events
+addHook({
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/cucumber.js:536]: ℹ️ only new hook for parallel mode

File: packages/datadog-plugin-cucumber/src/index.js [hunk 6]
```diff
@@ -213,10 +284,11 @@ class CucumberPlugin extends CiPlugin {
   }
 
   startTestSpan (testName, testSuite, extraTags) {
+    const testSuiteSpan = this.testSuiteSpanByPath[testSuite]
     return super.startTestSpan(
       testName,
       testSuite,
-      this.testSuiteSpan,
```
Comment (juan-fernandez) [inline packages/datadog-plugin-cucumber/src/index.js:219]: ℹ️ now there can be multiple test suite spans active at the same time, that's why we can use `this.testSuiteSpan`

File: packages/dd-trace/src/ci-visibility/exporters/test-worker/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
+'use strict'
```
Comment (juan-fernandez) [inline packages/dd-trace/src/ci-visibility/exporters/test-worker/index.js:1]: ℹ️ rename from `jest-worker` with some small changes for accommodating cucumber

File: packages/dd-trace/test/ci-visibility/exporters/test-worker/exporter.spec.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+'use strict'
```
Comment (juan-fernandez) [inline packages/dd-trace/test/ci-visibility/exporters/test-worker/exporter.spec.js:1]: ℹ️ renamed from `jest-worker` and added tests for cucumber

**Review by anmarchenko — APPROVED** [review]

**Review by rochdev — APPROVED** [review]

**Pull request merged**